### PR TITLE
Implement IAppxManifestReader5

### PR DESCRIPTION
**Pull request by stephenk-msft**

### CODE REVIEW

File: src/inc/AppxManifestObject.hpp [hunk 1]
```diff
@@ -232,13 +232,21 @@ namespace MSIX {
         HRESULT STDMETHODCALLTYPE GetMainPackageName(LPWSTR* mainPackageName) noexcept override try
         {
             ThrowErrorIf(Error::InvalidParameter, (mainPackageName == nullptr || *mainPackageName != nullptr), "bad pointer");
+            if (m_mainPackageName.empty())
+            {
+                return static_cast<HRESULT>(Error::OK);
+            }
```
Comment (msftrubengu) [inline src/inc/AppxManifestObject.hpp]: This is not necessary MarshalOutString already handles this logic

File: src/inc/AppxManifestObject.hpp [hunk 2]
```diff
@@ -232,13 +232,21 @@ namespace MSIX {
...
         } CATCH_RETURN();
 
         // IAppxManifestOptionalPackageInfoUtf8
         HRESULT STDMETHODCALLTYPE GetMainPackageName(LPSTR* mainPackageName) noexcept override try
         {
             ThrowErrorIf(Error::InvalidParameter, (mainPackageName == nullptr || *mainPackageName != nullptr), "bad pointer");
+            if (m_mainPackageName.empty())
+            {
+                return static_cast<HRESULT>(Error::OK);
+            }
```
Comment (msftrubengu) [inline src/inc/AppxManifestObject.hpp]: Same for MarshalOutStringUtf8

**Comment by msftrubengu** [comment]
> #include "Crypto.hpp"

Do you still need include these?

---
Refers to: src/msix/common/AppxPackageInfo.cpp:10 in 3e31ba8. [](commit_id = 3e31ba89ecda5071f3c2ef21b26e3e6c17f138b2, deletion_comment = False)

**Comment by stephenk-msft** [comment]
> > #include "Crypto.hpp"
> 
> Do you still need include these?
> 
> Refers to: src/msix/common/AppxPackageInfo.cpp:10 in [3e31ba8](https://github.com/microsoft/msix-packaging/commit/3e31ba89ecda5071f3c2ef21b26e3e6c17f138b2). [](commit_id = [3e31ba8](https://github.com/microsoft/msix-packaging/commit/3e31ba89ecda5071f3c2ef21b26e3e6c17f138b2), deletion_comment = False)

Removed.

File: src/inc/AppxPackageInfo.hpp [hunk 3]
```diff
@@ -85,6 +88,17 @@ namespace MSIX {
         return result;
     }
 
+    static std::string ComputePublisherId(const std::string& publisher)
+    {
+        auto wpublisher = utf8_to_u16string(publisher);
+        std::vector<std::uint8_t> buffer(wpublisher.size() * sizeof(char16_t));
+        memcpy(buffer.data(), &wpublisher[0], wpublisher.size() * sizeof(char16_t));
```
Comment (JohnMcPMS) [inline src/inc/AppxPackageInfo.hpp]: I would just use insert, but this will do.

**Pull request merged**